src/adapter/src/coord/sequencer/inner.rs

```diff
@@ -4001,6 +4001,25 @@ impl Coordinator {
         plan::AlterSystemSetPlan { name, value }: plan::AlterSystemSetPlan,
     ) -> Result<ExecuteResponse, AdapterError> {
         self.is_user_allowed_to_alter_system(session, Some(&name))?;
+        if NETWORK_POLICY.name.to_string().to_lowercase() == name.clone().to_lowercase() {
+            let values = match value {
+                plan::VariableValue::Default => None,
+                plan::VariableValue::Values(ref values) => Some(values),
+            };
+            if let Some(_policy) = values
+                .map(|v| self.catalog.get_network_by_name(&v[0]))
+                .flatten()
+            {
+            } else {
+                return Err(AdapterError::PlanError(plan::PlanError::VarError(
+                    VarError::InvalidParameterValue {
+                        name: NETWORK_POLICY.name(),
+                        invalid_values: values.expect("Must have provided value.").to_owned(),
+                        reason: "No network policy with such name exists!".to_string(),
+                    },
+                )));
+            }
```

Comment:
Do you mind refactoring this, it's a bit hard to grok at a glance, maybe something like:
```
let network_policy_exists = values
    .map(|v| self.catalog.get_network_policy_by_name(&v[0]))
    .flatten()
    .is_some();
if !network_policy_exists {
    ...
}
```
Also we generally allow folks to set system or session vars to an invalid value, e.g. you can set the `cluster` var to a non-existent cluster. I understand why we don't want to allow that here though, do you mind adding a comment calling out that this is a special case?

Reply:
By adding a comment do you mean to the definition or in some documentation?